stack-graphs/include/stack-graphs.h

```diff
@@ -104,6 +104,27 @@ struct sg_symbols {
 // handles using simple equality, without having to dereference them.
 typedef uint32_t sg_symbol_handle;
 
+// Arbitrary string content associated with some part of a stack graph.
+struct sg_string {
+    const char *content;
+    size_t length;
+};
+
+// An array of all of the interned strings in a stack graph.  String handles are indices into this
+// array. There will never be a valid string at index 0; a handle with the value 0 represents a
+// missing string.
+struct sg_strings {
+    const struct sg_string *strings;
+    size_t count;
+};
+
+// A handle to an interned string in a stack graph.  A zero handle represents a missing string.
+//
+// We deduplicate strings in a stack graph — that is, we ensure that there are never multiple
+// `struct sg_string` instances with the same content.  That means that you can compare string
+// handles using simple equality, without having to dereference them.
+typedef uint32_t sg_string_handle;
+
```

Comment:
We decided not to mark this as a generated file so that we can actually review its contents as part of PRs.  It's generated in the sense that a computer authors its contents, but it is intended for human consumption when people try to _use_ the C bindings as a library.  So I feel it's important to review the contents of the header file to make sure they look "right" for human consumption.

Reply:
Okay, that makes sense.